tedana/decay.py

```diff
@@ -97,6 +121,30 @@ def fit_decay(data, tes, mask, masksum):
         t2ss[..., echo - 1] = np.squeeze(utils.unmask(t2s, mask))
         s0vs[..., echo - 1] = np.squeeze(utils.unmask(s0, mask))
 
+    if fittype == 'curvefit':
+        # perform a monoexponential fit of echo times against MR signal
+        # using the mean of the signal, using loglin estimates
+        # as initial starting points for fit
+
+        t2scf = np.zeros(t2s.size)
+        s0cf = np.zeros(t2s.size)
+
+        for voxel in range(t2s.size):
+            try:
+                popt, cov = scipy.optimize.curve_fit(mono_exp, tes, fit_data[voxel, :],
+                                                     sigma=fit_sigma[voxel, :],
+                                                     p0=(s0[voxel],
+                                                     t2s[voxel]))
+                s0cf[voxel] = popt[0]
+                t2scf[voxel] = popt[1]
+            except RuntimeError:
```

Comment:
It would be nice if we had some kind of summary of how many voxels the curve-fitting method fail for.

Reply:
Agree - a counter would be easy enough, and maybe just logging it. If someone is using curvefit, showing the number of failing voxels(and percentage perhaps) for each loop shouldn't cause too much trouble.

Reply:
Added logging that reports the number and percentage of failed voxels per # of used echoes. Maybe this is overkill though, since it prints a line per loop...It is interesting, the number is much lower than I expected
`
INFO:tedana.decay:With 3 echoes, monoexponential fit failed on 32 (0.02%) voxel(s), used log linear estimate instead`
`INFO:tedana.decay:With 4 echoes, monoexponential fit failed on 29 (0.02%) voxel(s), used log linear estimate instead`

Reply:
Perhaps it's better suited as a debug-level message. It's definitely interesting (and informative) to see, but most users don't need that info. It's great for developers to have though!

Reply:
seems like it that is working nicely